radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBook.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.network.p2p.addressbook;
+
+import com.google.inject.Inject;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.network.p2p.NodeId;
+import com.radixdlt.network.p2p.RadixNodeUri;
+
+import javax.inject.Singleton;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.function.Predicate.not;
+
+/**
+ * Manages known peers network addresses and their metadata.
+ */
+@Singleton
+public final class AddressBook {
+	private static final Comparator<AddressBookEntry.PeerAddressEntry> entryComparator = (a, b) -> {
+		final var aLastSuccess = a.getLastSuccessfulConnection().orElse(Instant.MIN);
+		final var bLastSuccess = b.getLastSuccessfulConnection().orElse(Instant.MIN);
+		return -aLastSuccess.compareTo(bLastSuccess);
```

Comment:
Seems like this comparator basically ignores old entries? Or is there a way to get back to older entries?